### ISSUE TITLE
core-p7 branch introduces new build-time warnings

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#7]
Code changes in the core-p7 branch have introduced some build-time warnings not previously observed in blead.  At afb329e3853f12195063ad97276035824bfa4b4e, we get:
```
$ parse-build-warnings core-p7.linux.make-test-prep.output.txt.gz
File:  core-p7.linux.make-test-prep.output.txt.gz

[
  {
    char   => 32,
    group  => "Wincompatible-pointer-types",
    line   => 200,
    source => "IO.xs",
    text   => "passing argument 2 of \xE2\x80\x98PerlIO_getpos\xE2\x80\x99 from incompatible pointer type",
  },
  {
    char   => 34,
    group  => "Wincompatible-pointer-types",
    line   => 241,
    source => "IO.xs",
    text   => "passing argument 2 of \xE2\x80\x98PerlIO_setpos\xE2\x80\x99 from incompatible pointer type",
  },
  {
    char   => 2,
    group  => "Wdeprecated-declarations",
    line   => 247,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
  {
    char   => 2,
    group  => "Wdeprecated-declarations",
    line   => 292,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
  {
    char   => 2,
    group  => "Wdeprecated-declarations",
    line   => 369,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
  {
    char   => 2,
    group  => "Wdeprecated-declarations",
    line   => 639,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
  {
    char   => 2,
    group  => "Wdeprecated-declarations",
    line   => 676,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
  {
    char   => 2,
    group  => "Wdeprecated-declarations",
    line   => 734,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
  {
    char   => 6,
    group  => "Wdeprecated-declarations",
    line   => 743,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
  {
    char   => 6,
    group  => "Wdeprecated-declarations",
    line   => 774,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
  {
    char   => 2,
    group  => "Wdeprecated-declarations",
    line   => 913,
    source => "Normalize.xs",
    text   => "\xE2\x80\x98Perl_utf8n_to_uvuni\xE2\x80\x99 is deprecated",
  },
]
```
In addition to the above, when I run `make` on this branch on FreeBSD-11, immediately prior to the failure yesterday in https://github.com/atoomic/perl/issues/5, we (not surprisingly) get these two additional build-time warnings:
```
  {
    char   => 1,
    group  => "Wmissing-field-initializers",
    line   => 427,
    source => "threads.xs",
    text   => "missing field 'svt_local' initializer",
  },
  {
    char   => 1,
    group  => "Wmissing-field-initializers",
    line   => 427,
    source => "threads.xs",
    text   => "missing field 'svt_local' initializer",
  },
```
Thank you very much.
Jim Keenan

**Comment by atoomic** [comment]
very useful information and probably points to the issue in #5

**Comment by atoomic** [comment]
c1150af4859d7078dcefbfc3fa4fa9bc29ea14bc starts fixing issues from this case.
IO.xs should now be fixed with this change

**Comment by atoomic** [comment]
431a3015 should fix issues from Normalize.xs
threads.xs issues should be fixed by #5 

I think all described issues in this case are now resolved

**Comment by jkeenan** [comment]
> [431a301](https://github.com/atoomic/perl/commit/431a30153ffeb23a27a0be71440916aa2b41c819) should fix issues from Normalize.xs
> threads.xs issues should be fixed by #5
> 
> I think all described issues in this case are now resolved

Agreed.
```
$ ~/bin/perl/report-build-warnings 2d033b45d0.freebsd.unthreaded.make.test.prep.output.txt.gz 
File:  2d033b45d0.freebsd.unthreaded.make.test.prep.output.txt.gz

[no warnings reported]

$ ~/bin/perl/report-build-warnings 2d033b45d0.freebsd.threaded.make.test.prep.output.txt.gz 
File:  2d033b45d0.freebsd.threaded.make.test.prep.output.txt.gz

[no warnings reported]
```
Closing.  Thank you very much.
